consensus/hotstuff/pacemaker/pacemaker_test.go

```diff
@@ -30,456 +32,188 @@ func expectedTimerInfo(view uint64, mode model.TimeoutMode) interface{} {
+
+	s.notifier.AssertExpectations(s.T())
+	require.Equal(s.T(), uint64(13), s.paceMaker.CurView())
 }
 
-// Test_SkipViewThroughBlock tests that PaceMaker skips View when receiving Block containing QC with larger View Number
-//func Test_SkipViewThroughBlock(t *testing.T) {
-//	pm, notifier := initPaceMaker(t, 3)
-//
-//	block := makeBlock(5, 9)
-//	notifier.On("OnStartingTimeout", expectedTimerInfo(6, model.ReplicaTimeout)).Return().Once()
-//	notifier.On("OnQcTriggeredViewChange", block.QC, uint64(6)).Return().Once()
-//	nve, nveOccurred := pm.UpdateCurViewWithBlock(block, true)
-//	notifier.AssertExpectations(t)
-//	assert.Equal(t, uint64(6), pm.CurView())
-//	assert.True(t, nveOccurred && nve.View == 6)
-//
-//	block = makeBlock(22, 25)
-//	notifier.On("OnStartingTimeout", expectedTimerInfo(23, model.ReplicaTimeout)).Return().Once()
-//	notifier.On("OnQcTriggeredViewChange", block.QC, uint64(23)).Return().Once()
-//	nve, nveOccurred = pm.UpdateCurViewWithBlock(block, false)
-//	assert.True(t, nveOccurred && nve.View == 23)
-//
-//	notifier.AssertExpectations(t)
-//	assert.Equal(t, uint64(23), pm.CurView())
-//}
-
-// Test_HandlesSkipViewAttack verifies that PaceMaker skips views based on QC.view
-// but NOT based on block.View to avoid vulnerability against Fast-Forward Attack
-//func Test_HandlesSkipViewAttack(t *testing.T) {
-//	pm, notifier := initPaceMaker(t, 3)
-//
-//	block := makeBlock(5, 9)
-//	notifier.On("OnStartingTimeout", expectedTimerInfo(6, model.ReplicaTimeout)).Return().Once()
-//	notifier.On("OnQcTriggeredViewChange", block.QC, uint64(6)).Return().Once()
-//	nve, nveOccurred := pm.UpdateCurViewWithBlock(block, true)
-//	notifier.AssertExpectations(t)
-//	assert.Equal(t, uint64(6), pm.CurView())
-//	assert.True(t, nveOccurred && nve.View == 6)
-//
-//	block = makeBlock(14, 23)
-//	notifier.On("OnStartingTimeout", expectedTimerInfo(15, model.ReplicaTimeout)).Return().Once()
-//	notifier.On("OnQcTriggeredViewChange", block.QC, uint64(15)).Return().Once()
-//	nve, nveOccurred = pm.UpdateCurViewWithBlock(block, false)
-//	assert.True(t, nveOccurred && nve.View == 15)
-//
-//	notifier.AssertExpectations(t)
-//	assert.Equal(t, uint64(15), pm.CurView())
-//}
-
-// Test_IgnoreOldBlocks tests that PaceMaker ignores old blocks
-//func Test_IgnoreOldBlocks(t *testing.T) {
-//	pm, notifier := initPaceMaker(t, 3)
-//	pm.UpdateCurViewWithBlock(makeBlock(1, 2), false)
-//	pm.UpdateCurViewWithBlock(makeBlock(1, 2), true)
-//	notifier.AssertExpectations(t)
-//	assert.Equal(t, uint64(3), pm.CurView())
-//}
+// TestProcessQC_PersistException tests that ActivePaceMaker propagates exception
+// when processing QC
+func (s *ActivePaceMakerTestSuite) TestProcessQC_PersistException() {
+	exception := errors.New("persist-exception")
+	qc := QC(3)
+	s.persist.On("PutLivenessData", mock.Anything).Return(exception).Once()
+	nve, err := s.paceMaker.ProcessQC(qc)
+	require.Nil(s.T(), nve)
+	require.ErrorAs(s.T(), err, &exception)
```

Comment:
```suggestion
	require.ErrorIs(s.T(), err, exception)
```
Better to use the more specific error equality assertion